packages/sdk/src/oracle/services/sortedRequests.ts

```diff
@@ -1,16 +1,102 @@
-import { SortedJsMap } from "../../stores";
-import { RequestIndex, RequestIndexes } from "../types/state";
+import sortedIndex from "lodash/sortedIndex";
+import sortedLastIndex from "lodash/sortedLastIndex";
 
-export class SortedRequests {
-  private table = SortedJsMap<string, RequestIndex>();
-  async set(key: string, value: RequestIndex): Promise<void> {
-    await this.table.set(key, value);
+import { exists } from "../../utils";
+import { requestId } from "../utils";
+
+import { RequestIndex, RequestIndexes, InputRequest } from "../types/state";
+
+// this was copied out of store and made to be 1. a class, and 2. sync.  Could not modify old implementation since
+// other services rely on this currently and it causes cascading refactors. This has been copied without modification to logic.
+export class SortedStore<Id, Data> {
+  private ids: Id[] = [];
+  private map = new Map<Id, Data>();
+  private getStart = (id: Id, inclusive = true): number => {
+    if (inclusive) {
+      return sortedIndex(this.ids, id);
+    } else {
+      return sortedLastIndex(this.ids, id);
+    }
+  };
+  private getEnd = (id: Id, inclusive = false): number => {
+    return this.getStart(id, !inclusive);
+  };
+  private del = (id: Id): Data | undefined => {
+    const index = sortedIndex(this.ids, id);
+    this.ids.splice(index, 1);
+    const result = this.map.get(id);
+    this.map.delete(id);
+    return result;
+  };
+  // keeps typescript from complaining that return values may be undefined
+  private getSafe = (id: Id): Data => {
+    const result = this.map.get(id);
+    if (exists(result)) return result;
+    throw new Error("Missing data for index: " + id);
+  };
+  has = (id: Id): boolean => {
+    return this.map.has(id);
+  };
+  set = (id: Id, data: Data): void => {
+    if (this.map.has(id)) {
+      this.map.set(id, data);
+    } else {
+      const index = sortedIndex(this.ids, id);
+      this.ids.splice(index, 0, id);
+      this.map.set(id, data);
+    }
+  };
+  get = (id: Id): Data | undefined => {
+    return this.map.get(id);
+  };
+  values = (): Data[] => {
+    return this.ids.map(this.getSafe);
+  };
+  entries = (): [Id, Data][] => {
+    return this.ids.map((id) => [id, this.getSafe(id)]);
+  };
+  keys = (): Id[] => {
+    return [...this.ids];
+  };
+  clear = (): void => {
+    this.map.clear();
+    this.ids.length = 0;
+  };
+  size = (): number => {
+    return this.ids.length;
+  };
+  delete = (id: Id): void => {
+    if (!this.map.has(id)) return;
+    this.del(id);
+  };
+  // assume [a,b)
+  between = (a: Id, b: Id): Data[] => {
+    const start = this.getStart(a);
+    const end = this.getEnd(b);
+    return this.ids.slice(start, end).map(this.getSafe);
+  };
+  slice = (id: Id, length: number): Data[] => {
+    const start = this.getStart(id);
+    return this.ids.slice(start, start + length).map(this.getSafe);
+  };
+}
```

Comment:
this is literally copied and translated from /stores/sorted-js-map/store.ts. Perhaps this can eventually replace the original, but previously ran into issues when converting that to a class, it would have cascading effects on the rest of the code in the sdk and api. So decided to move it locally to this client for now.